### ISSUE TITLE
Implement "draft" review API for commenting

### ISSUE THREAD

**Issue by EnTeQuAk** [mozilla/addons-server#11277]
As part of https://github.com/mozilla/addons-server/issues/10621 and https://github.com/orgs/mozilla/projects/79 we need some way for the reviewers to prefill the "comment" section during a review in the reviewer-tools.

The idea is to have a "draft"-mode, represented by a new model that saves comments or any potential activity that the code-manager sends to the reviewer-api which then allows the reviewer to simply hit the "submit" button during a review in the reviewer-tools.

Reviewers will have to use two tools for a long time anyway so that's already simplifying their live a bit.

A reviewer will then be able to comment on files and diffs in the code-manager and do so inline, switch back to the reviewer-tools, review his actions and hit submit.

**Comment by EnTeQuAk** [comment]
Some more details from what we discussed:

We decided that the frontend pull together all the comments and submits them as one to the API. This will most certainly change eventually to allow better stats tracking etc once we know how things are working with inline-comments.

**Comment by willdurand** [comment]
> We decided that the frontend pull together all the comments and submits them as one to the API. This will most certainly change eventually to allow better stats tracking etc once we know how things are working with inline-comments.

From https://docs.google.com/document/d/1y4crrHxhGejb3eZ6HpjtBdDqONJakfwtfjaOqwq-qZo/edit#,it looks like the API needs to accept a list of comments and build the draft itself. Did I miss anything?

**Comment by EnTeQuAk** [comment]
> From docs.google.com/document/d/1y4crrHxhGejb3eZ6HpjtBdDqONJakfwtfjaOqwq-qZo/edit#,it looks like the API needs to accept a list of comments and build the draft itself. Did I miss anything?

No, you're correct. This is the very first step to get some real-world experimentation and something working. The next steps in my mind would be:

* Expose list of canned responses
* Allow storing a list of comments (optionally linking to a canned response) linking to a specific line (optionally a range of lines)

Splitting the implementation up like this in smaller parts allows us to check in more regularly and adapt accordingly to how the implementation in the frontend goes so that feedback can be incorporated into api development easier. At least, that's my dream-world :see_no_evil:

**Comment by willdurand** [comment]
Great, thanks!

**Comment by AlexandraMoga** [comment]
@EnTeQuAk I've read through the discussions you had around this new implementation and tried to figure out a test plan.

The following observation also applies to #11319 and #11386 :
If I understand correctly, the new [draft comments API](https://addons-server.readthedocs.io/en/latest/topics/api/reviewers.html#drafting-comments) should be able to store reviewer comments entered through the code-manager until the full review is finally submitted from Rev Tools. However, it looks like without https://github.com/mozilla/addons-server/issues/11378 I won't be able to achieve any practical testing around this feature. If there is a workaround,  please let me know.
